### DISCUSSION TITLE
How to store and retrieve the anchor from local storage.

### DISCUSSION THREAD

**Discussion by sagarjgborg** [sceneview/sceneview#362] [Q&A]
I am building an application in which users can store their added anchors in a plane in local storage and retrieve them again when required. Is there any way to do so?

Thanks in advance.

**Comment by ThomasGorisse** [comment]
ARCore anchors are session-specific and can't be directly serialized to storage. For persistent anchors across sessions, use **ARCore Cloud Anchors**:

```kotlin
ARSceneView(
    sessionConfiguration = { session, config ->
        config.cloudAnchorMode = Config.CloudAnchorMode.ENABLED
    }
) {
    // Host an anchor
    session.hostCloudAnchorAsync(localAnchor, 365) { cloudAnchorId ->
        // Save cloudAnchorId to local storage / database
    }
    
    // Resolve it later
    session.resolveCloudAnchorAsync(savedCloudAnchorId) { anchor ->
        // Use the resolved anchor
    }
}
```

Cloud Anchors let you save and restore anchor positions across sessions and devices. You store the cloud anchor ID (a string) and resolve it later.
